airflow/ti_deps/deps/not_previously_skipped_dep.py

```diff
@@ -49,46 +48,47 @@ def _get_dep_statuses(self, ti, session, dep_context):
         finished_task_ids = {t.task_id for t in finished_tis}
 
         for parent in upstream:
-            if isinstance(parent, SkipMixin):
```

Comment:
Removing this makes the scheduler have to check this for all operators where it didn't before. 

Given this is in the hottest of hot paths in the scheduler we might need to rethink this: perhaps we store a flag in the ser dag like we did for `is_empty`

Reply:
If we retain it - we also need to check whether it's a [mapped operator](https://github.com/apache/airflow/pull/44925#discussion_r1884624755), otherwise we have issues in that area

Reply:
Implemented the flag as we discussed.
When working on the tests I'll check that there are no issues with mapped operators.